example/android/app/src/main/java/com/example/MainApplication.java

```diff
@@ -55,7 +55,11 @@ public ReactNativeHost getReactNativeHost() {
   @Override
   public void onCreate() {
     super.onCreate();
-    FirebaseApp.initializeApp(this);
+    
```

Comment:
I assume based on your earlier comment you plan to uncomment this and use the enablePushNotification flag here to conditionally init. That sounds like a good plan to me.

Reply:
Probably not worth the effort right now but there are ways to dynamically inject dependencies using Dagger. For now, uncommenting is easy enough but if we want to clean this up in the future we could consider an approach like that.

Reply:
sweet! yeah i was trying to dig in to a way to use dynamic imports in java but not much progress on that front. Dagger sounds cool!